nvtabular/tools/data_gen.py

```diff
@@ -0,0 +1,442 @@
+                entrys_lens = dist.create_col(
+                    col_size + 1, dtype=np.long, min_val=mh_min, max_val=mh_max
+                ).ceil()
+                # sum returns numpy dtype
+                col_size = int(entrys_lens.sum())
+                offs = cupy.cumsum(entrys_lens.values)
+            ser = dist.create_col(col_size, dtype=np.long, min_val=0, max_val=cardinality).ceil()
+            if entries:
+                cat_names = self.create_cat_entries(cardinality, min_size=minn, max_size=maxx)
+                ser, _ = self.merge_cats_encoding(ser, cat_names)
+            if offs is not None:
+                # create multi_column from offs and ser
+                ser = self.create_multihot_col(offs, ser)
+            ser.name = name
+            df = cudf.concat([df, ser], axis=1)
+        return df
+
+    def create_labels(self, size, labs_rep):
+        num_cols = len(labs_rep)
+        df = cudf.DataFrame()
+        for x in range(num_cols):
+            name, dtype, distro, cardinality = labs_rep[x][:4]
+            dist = distro or self.dist
+            ser = dist.create_col(size, dtype=np.long, min_val=1, max_val=cardinality).ceil()
+            # bring back down to correct representation because of ceil call
+            ser = ser - 1
+            ser.name = name
+            df = cudf.concat([df, ser], axis=1)
+        return df
+
+    def merge_cats_encoding(self, ser, cats):
+        # df and cats are both series
+        # set cats to dfs
+        offs = None
+        if type(ser.dtype) is cudf.core.dtypes.ListDtype:
+            offs = ser._column.offsets
+            ser = ser.list.leaves
+        ser = cudf.DataFrame({"vals": ser})
+        cats = cudf.DataFrame({"names": cats})
+        cats["vals"] = cats.index
+        ser = ser.merge(cats, on=["vals"], how="left")
+        return ser["names"], offs
+
+    def create_cat_entries(self, cardinality, min_size=1, max_size=5):
+        set_entries = []
+        while len(set_entries) <= cardinality:
+            letters = string.ascii_letters + string.digits
+            entry_size = random.randint(min_size, max_size)
+            entry = "".join(random.choice(letters) for i in range(entry_size))
+            if entry not in set_entries:
+                set_entries.append(entry)
+        return set_entries
+
+    def create_multihot_col(self, offsets, data):
+        """
+        offsets = cudf series with offset values for list data
+        data = cudf series with the list data flattened to 1-d
+        """
+        offs = as_column(offsets, dtype="int32")
+        encoded = as_column(data)
+        col = build_column(
+            None,
+            size=offs.size - 1,
+            dtype=cudf.core.dtypes.ListDtype(encoded.dtype),
+            children=(offs, encoded),
+        )
+        return cudf.Series(col)
+
+    def create_df(
+        self,
+        size,
+        conts_rep=None,
+        cats_rep=None,
+        labs_rep=None,
```

Comment:
Can we simplify the API a bit here? Rather than pass in conts_rep/cats_rep/labs_rep - can we just pass in the dataset schema and then call _get_cols_from_schema to get those 3?